Napzak-iOS/Napzak-iOS/Presentation/Register/Components/RegisterImage.swift

```diff
@@ -20,25 +28,148 @@ struct RegisterImage: View {
+
+                    }
+                    .frame(width: 88, height: 88)
+                    .background(Color.napzakPrimary(.purple100))
+                    .clipShape(.rect(cornerRadius: 5))
+                    .padding(.trailing, 14)
+                }
+                .disabled(disabled)
                 
-                Text("사진 0/10")
-                    .applyNapzakFont(.caption3Regular12)
-                    .foregroundStyle(Color.napzakPrimary(.purple500))
-                    .frame(height: 13)
+                ForEach(0..<selectedImages.count, id: \.self) { index in
+                    imageItemView(for: index)
+                        .padding(.trailing, 8)
+                }
             }
-            .frame(width: 88, height: 88)
-            .background(Color.napzakPrimary(.purple100))
-            .clipShape(.rect(cornerRadius: 5))
             
         }
+        .frame(height: 96)
+    }
+    
+    private var representativeBadge: some View {
+        VStack(alignment: .center) {
+            Text("대표")
+                .applyNapzakFont(.caption5Regular10)
+                .foregroundStyle(.white)
+                .frame(height: 13)
+        }
+        .frame(width: 46, height: 23)
+        .background(Color.napzakTransparency(.transBlack))
+        .clipShape(
+            .rect(
+                topLeadingRadius: 5,
+                bottomTrailingRadius: 5
+            )
+        )
     }
 }
 
-#Preview {
-    RegisterImage()
+
+// MARK: - Functions
+
+extension RegisterImage {
+    
+    private func imageItemView(for index: Int) -> some View {
+        ZStack(alignment: .bottomLeading) {
+            Image(uiImage: selectedImages[index])
+                .resizable()
+                .scaledToFill()
+                .frame(width: 88, height: 88)
+                .clipShape(RoundedRectangle(cornerRadius: 5))
+                .overlay(alignment: .topLeading) {
+                    if index == 0 {
+                        representativeBadge
+                    }
+                }
+            deleteButton(at: index)
+        }
+        .frame(width: 95, height: 96)
+    }
+    
+    private func deleteButton(at index: Int) -> some View {
+        VStack{
+            HStack{
+                Spacer()
+                Image(.iconImageCancle)
+                    .frame(width: 16, height: 16)
+                    .onTapGesture {
+                        print("xbutton tapped")
+                        selectedImages.remove(at: index)
+                        imageNameList.remove(at: index)
+                    }
+            }
+            Spacer()
+            Rectangle()
+                .fill(.gray.opacity(0.000000000000000000001))
+                .frame(width: 88)
+                .frame(maxHeight: .infinity)
+                .onLongPressGesture(perform: {
+                    print("picture long pressed")
+                    moveImageToFront(at: index)
+                })
```

Comment:
P2

```suggestion
            Color.clear
                .frame(width: 88)
                .frame(maxHeight: .infinity)
                .contentShape(Rectangle())
                .onLongPressGesture(perform: {
                    print("picture long pressed")
                    moveImageToFront(at: index)
                })
```

Color.clear에 contentShape을 잡아줘도 제스처를 넣을 수 있다는 사아실

Reply:
와 미쳤다

Reply:
또 하나 배웁니다.